packages/app/contexts/StackboxFormContext.tsx

```diff
@@ -0,0 +1,178 @@
+"use client";
+
+import { createContext, useContext, ReactNode, useMemo } from "react";
+
+import {
+  createParser,
+  parseAsInteger,
+  parseAsString,
+  parseAsStringEnum,
+  parseAsTimestamp,
+  useQueryState,
+} from "next-usequerystate";
+import { useNetwork } from "wagmi";
+
+import {
+  TokenWithBalance,
+  useStrategyContext,
+  useTokenListContext,
+} from "@/contexts";
+import { FREQUENCY_OPTIONS } from "@/models/stack";
+import { ChainId } from "@stackly/sdk";
+import { DEFAULT_TOKENS_BY_CHAIN } from "@/utils/constants";
+
+const endDateByFrequency: Record<string, number> = {
+  [FREQUENCY_OPTIONS.hour]: new Date().setDate(new Date().getDate() + 2),
+  [FREQUENCY_OPTIONS.day]: new Date().setMonth(new Date().getMonth() + 1),
+  [FREQUENCY_OPTIONS.week]: new Date().setMonth(new Date().getMonth() + 3),
+  [FREQUENCY_OPTIONS.month]: new Date().setFullYear(
+    new Date().getFullYear() + 1
+  ),
+};
+
+const startDateParseAsTimestamp = createParser({
+  parse: (v) => {
+    const ms = parseInt(v);
+    if (Number.isNaN(ms)) {
+      return null;
+    }
+
+    const searchParamsStartDate = new Date(ms);
+    const nowDate = new Date();
+
+    if (nowDate.getTime() > searchParamsStartDate.getTime()) return nowDate;
+
+    return new Date(ms);
+  },
+  serialize: (v: Date) => v.valueOf().toString(),
+});
+
+const throwStackboxFormContextError = () => {
+  throw new Error("No StackboxFormContext available");
+};
+
+/**
+ * @TODO enhance `stackboxFormState` typing
+ * @see https://github.com/youha-info/next-query-state#background-on-types
+ * @see https://www.typescriptlang.org/docs/handbook/2/generics.html
+ */
+interface StackboxFormContextProps {
+  resetDefaultFormValues: () => void;
+  stackboxFormState: any;
+}
+
+const StackboxFormContext = createContext<StackboxFormContextProps>({
+  resetDefaultFormValues: throwStackboxFormContextError,
```

Comment:
I would rename it to `resetFormValues`, I think reset is enough to intent that it goes to reset state.

```suggestion
  resetFormValues: throwStackboxFormContextError,
```